mbtiles/scripts/cli.py

```diff
@@ -113,98 +132,119 @@ def mbtiles(ctx, files, output, force_overwrite, title, description,
+
+            # Initialize the sqlite db.
+            if os.path.exists(output):
+                os.unlink(output)
+            conn = sqlite3.connect(output)
+            cur = conn.cursor()
+            cur.execute(
+                "CREATE TABLE tiles "
+                "(zoom_level integer, tile_column integer, "
+                "tile_row integer, tile_data blob);")
+            cur.execute(
+                "CREATE TABLE metadata (name text, value text);")
 
-            # Insert tile into db.
+            # Insert mbtiles metadata into db.
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("name", title))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("type", layer_type))
             cur.execute(
-                "INSERT INTO tiles "
-                "(zoom_level, tile_column, tile_row, tile_data) "
-                "VALUES (?, ?, ?, ?);",
-                (tile.z, tile.x, tiley, buffer(contents)))
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("version", "1.1"))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("description", description))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("format", img_ext))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("bounds", "%f,%f,%f,%f" % (west, south, east, north)))
 
             conn.commit()
 
+            # Constrain longitude and latitude bounds.
+            EPS = 1.0e-10
+            west = max(-180 + EPS, west)
+            south = max(-85.051129, south)
+            east = min(180 - EPS, east)
+            north = min(85.051129, north)
+
+            with WarpedVRT(
+                    src, dst_crs='EPSG:3857', src_nodata=src_nodata,
+                    dst_nodata=dst_nodata, resampling=Resampling.bilinear,
+                    num_threads=num_workers) as vrt, warnings.catch_warnings():
+
+                warnings.simplefilter("ignore", category=UserWarning)
+
+                for tile in mercantile.tiles(
+                        west, south, east, north,
+                        list(range(minzoom, maxzoom + 1))):
+
+                    window = vrt.window(*mercantile.xy_bounds(*tile))
+                    bands = vrt.read(
+                        window=window, out_shape=(3, 256, 256),
+                        boundless=True)
+
+                    with MemoryFile() as memfile:
+                        # Silence the dubious 'No such file or directory.'
+                        # errors coming from the VSIL handler.
+                        gdal_log = logging.getLogger('rasterio._gdal')
+                        gdal_log_level = gdal_log.level
+                        gdal_log.setLevel(logging.CRITICAL)
```

Comment:
There's a dubious error raised by GDAL when creating a new `/vsimem` dataset, I guess from trying to open it in read mode first? Anyhow, this is a way to silence that error.

Reply:
Upstream ticket here: https://trac.osgeo.org/gdal/ticket/7150.